src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -856,6 +866,37 @@ private async Task<DownloadedBuild> GatherDropForBuildAsync(Build build, string
                     };
                 }
             }
+            else if (IsAzureDevOpsFeedUrl(assetLocation.Location, out var feedAccount, out var feedVisibility, out var feedName))
+            {
+                string packageContentUrl = $"https://pkgs.dev.azure.com/{feedAccount}/{feedVisibility}_apis/packaging/feeds/{feedName}/nuget/packages/{asset.Name}/versions/{asset.Version}/content";
+
+                string fullTargetPath = Path.Combine(subPath, packagesSubPath, $"{asset.Name}.{asset.Version}.nupkg");
+
+                // feedVisibility == "" means that the feed is internal.
+                if (string.IsNullOrEmpty(feedVisibility))
+                {
+                    if (string.IsNullOrEmpty(_options.AzureDevOpsPat))
+                    {
+                        LocalSettings localSettings = LocalSettings.LoadSettingsFile(_options);
```

Comment:
How does the exception look when it's bubbled up when there's no PAT in the options, or in the settings file?

Reply:
If the PAT is not available, we should not add authorization headers.

If the feed is private, it will; say it's unauthorized.  That exception gets caught and shown to the user.
If the feed is public, it will just work.

Reply:
> If the PAT is not available, we should not add authorization headers.
> 
> If the feed is private, it will; say it's unauthorized. That exception gets caught and shown to the user.
> If the feed is public, it will just work.

AHh that is what you are doing, though I think the feed visibility is "public" or empty

Reply:
It will catch in the outer exception handler as an unauthorized access exception.